### PR TITLE
Enabled loop exit structurizer by default.

### PR DESCRIPTION
**Pull request by adam-yang**
Also:
- Replaced undefs with 0's for numeric types.
- Fixed a bug in loop exit structurizer where exit values don't dominate users.

### CODE REVIEW

File: lib/Transforms/Scalar/DxilRemoveUnstructuredLoopExits.cpp [hunk 1]
```diff
@@ -297,21 +307,25 @@ static void SkipBlockWithBranch(BasicBlock *bb, Value *cond, Loop *L, LoopInfo *
...
     PHINode *phi = nullptr;
 
-    for (User *user : inst.users()) {
-      Instruction *user_inst = dyn_cast<Instruction>(user);
-      if (!user_inst)
-        continue;
+    // Put the users in a list, since we're modifying the user list as we go
+    users.clear();
+    for (User *user : inst.users())
+      users.push_back(user);
```
Comment (pow2clk) [inline lib/Transforms/Scalar/DxilRemoveUnstructuredLoopExits.cpp:317]: I think you could also use the iterator_range constructor of SmallVector here to the same effect and slightly simpler code.

```c++
SmallVector<User *,4> users(inst.users());
```

However, I think the more common approach in this codebase to this problem is something like this replacing the previous loop:

```c++
for (auto I = inst.users().begin(), E = inst.users().end(); I != E; ++I) {
  Instruction *user_inst = dyn_cast<Instruction>(*I);
...
```

Which I think would have the same effect.

It's not terribly important. What's here is fine. I just wanted to point out alternatives you might consider.

**Review by pow2clk — COMMENTED** [review]
Looks good overall. I'll give tex a chance to weigh in. Meantime I have some alternatives you might consider for the loop problem.

File: lib/Transforms/Scalar/DxilRemoveUnstructuredLoopExits.cpp (hunk 1 above)
Comment (adam-yang) [reply lib/Transforms/Scalar/DxilRemoveUnstructuredLoopExits.cpp]: Doh! Thanks for pointing it out.

**Review by pow2clk — APPROVED** [review]
Looks good to me!

**Pull request merged**